File: from-0501-to-0600/0506-relative-ranks/solution.go
func findRelativeRanks(score []int) []string {
    idx := make(map[int]int)
    for i := 0; i < len(score); i ++ {
        idx[score[i]] = i
    }
    
    sort.Slice(score, func(i, j int) bool {
        return score[i] > score[j]
    })

    result := make([]string, len(score))
    for i := 0; i < len(score); i ++ {
        if i == 0 {
            result[idx[score[i]]] = "Gold Medal"
        } else if i == 1 {
            result[idx[score[i]]] = "Silver Medal"
        } else if i == 2 {
            result[idx[score[i]]] = "Bronze Medal"
        } else {
            result[idx[score[i]]] = strconv.Itoa(i + 1)
        }
    }
    return result
}

File: from-0501-to-0600/0506-relative-ranks/solution.py
class Solution:
    def findRelativeRanks(self, score: List[int]) -> List[str]:
        d_scores = {s: i for i, s in enumerate(score)}
        ans = [""] * len(score)
        for p, s in enumerate(sorted(score, reverse=True), start=1):
            place = str(p)
            if p == 1:
                place = "Gold Medal"
            elif p == 2:
                place = "Silver Medal"
            elif p == 3:
                place = "Bronze Medal"
                
            ans[d_scores[s]] = place
            
        return ans